README.md
# tmgdev


tcpdf_old/CHANGELOG.TXT
5.9.149 (2012-02-21)
	- Bug item #3489933 "SVG Parser treats tspan like text" was fixed.

5.9.148 (2012-02-17)
	- Bug item #3488600 "Multiple radiobutton sets get first set value" was fixed.

5.9.147 (2012-02-14)
	- A problem with SVG gradients has been fixed.

5.9.146 (2012-02-12)
	- Bug item #3486880 "$filehash undefine error" was fixed.
	- The default font is now the one specified at PDF_FONT_NAME_MAIN constant.

5.9.145 (2012-01-28)
	- Japanese language file was added.
	- TCPDF license and README.TXT files were updated.

5.9.144 (2012-01-12)
	- HTML output on barcode classes was improved.

5.9.143 (2012-01-08)
	- Bug item #3471057 "setCreator() has no effect" was fixed.

5.9.142 (2011-12-23)
	- Source code documentation was updated.

5.9.141 (2011-12-14)
	- Some minor bugs were fixed.

5.9.140 (2011-12-13)
	- SVG now supports embedded images encoded as base64.

5.9.139 (2011-12-11)
	- Spot color methods were fixed.

5.9.138 (2011-12-10)
	- cropMark() method was improved (check source code documentation).
	- Example n. 56 was updated.
	- Bug item #3452390 "Check Box still not ticked when set to true" was fixed.

5.9.137 (2011-12-01)
	- Bug item #3447005 "Background color and border of Form Elements is printed" was fixed.
	- Color support for Form elements was improved.

5.9.136 (2011-11-27)
	- Bug item #3443387 "SetMargins with keep option does not work for top margin" was fixed.

5.9.135 (2011-11-04)
	- Bug item #3433406 "Double keywords in description" was fixed.

5.9.134 (2011-10-29)
	- The default value for $defcol parameter on convertHTMLColorToDec() method was fixed.
	- Deafult HTTP headers were changed to avoid browser caching.
	- Some deprecated syntax were replaced.

5.9.133 (2011-10-26)
	- Bug item #3428446 "copyPage method not working when diskcache enabled" was fixed.

5.9.132 (2011-10-20)
	- Bug item #3426167 "bug in function convertHTMLColorToDec()" was fixed.

5.9.131 (2011-10-13)
	- An error message was added to ImagePngAlpha() method.

5.9.130 (2011-10-12)
	- Now you can set image data strings on HTML img tag by encoding the image binary data in this way: $imgsrc = '@'.base64_encode($imgdata);

5.9.129 (2011-10-07)
	- Core fonts metrics was fixed (replace all helvetica and times php files on fonts folder).
	- Form fields support was improved and some problems were fixed (check the example n. 14).
	- Bug item #3420249 "Issue with booklet and MultiCell" was fixed.

5.9.128 (2011-10-06)
	- Method addTTFfont() was improved (check the source code documentation).
	- Method setExtraXMP() to set custom XMP data was added.

5.9.127 (2011-10-04)
	- Readonly mode option was activated for radiobuttons.

5.9.126 (2011-10-03)
	- Bug item #3417989 "Graphics State operator in form XObject fails to render" was fixed.
	- Xobjects problems with transparency, gradients and spot colors were fixed.

5.9.125 (2011-10-03)
	- Support for 8-digit CMYK hexadecimal color representation was added (to be used with XHTML and SVG).
	- Spot colors support was improved (check example n. 37).
	- Color methods were improved.

5.9.124 (2011-10-02)
	- Core fonts were updated.

5.9.123 (2011-10-02)
	- The method addTTFfont() wad added to automatically convert TTF fonts (check the new fonts guide at http://www.tcpdf.org).
	- Old font utils were removed.
	- All fonts were updated and new arabic fonts were added (almohanad were removed and replaced by aefurat and aealarabiya).
	- The file unicode_data.php was updated.
	- The file encodings_maps.php was added.
	- PDF/A files are now compressed to save space.
	- XHTML input form fields now support text-alignment attribute.

5.9.122 (2011-09-29)
	- PDF/A-1b compliance was improved to pass some online testing.

5.9.121 (2011-09-28)
	- This version includes support for PDF/A-1b format (the class constructor signature was changed - see example n. 65).
	- Method setSRGBmode() was added to force sRGB_IEC61966-2.1 black scaled ICC color profile for the whole document (file sRGB.icc was added).
	- 14 new fonts were added to allow embedding core fonts (for PDF/A compliance).
	- Font utils were fixed.

5.9.120 (2011-09-22)
	- This version includes a fix for _getTrueTypeFontSubset() method.

5.9.119 (2011-09-19)
	- This version includes a fix for extra page numbering on TOC.

5.9.118 (2011-09-17)
	- This version includes some changes that allows you to add a bookmark for a page that do not exist.

5.9.117 (2011-09-15)
	- TCPDFBarcode and TCPDF2DBarcode classes were extended to include a method for exporting barcodes as PNG images.

5.9.116 (2011-09-14)
	- Datamatrix class was improved and documentation was fixed.

5.9.115 (2011-09-13)
	- Datamatrix ECC200 barcode support was added (a new datamatrix.php file was added) - check example n. 50.
	- getBarcodeHTML() method was added on TCPDFBarcode and TCPDF2DBarcode classes to return an HTML representation of the barcode.
	- cURL options on Image() method were improved.
	- A bug on write2DBarcode() was fixed.

5.9.114 (2011-09-04)
	- A bug related to column position was fixed.

5.9.113 (2011-08-24)
	- This release include two new experimental files for parsing an existing PDF document (the integration with TCPDF is under development).

5.9.112 (2011-08-18)
	- A newline character was added after the 'trailer' keyword for compatibility with some parsers.
	- Support for layers was improved.

5.9.111 (2011-08-17)
	- Barcode CODE 39 default gap was restored at 1.

5.9.110 (2011-08-17)
	- Barcode CODE 39 was fixed.

5.9.109 (2011-08-12)
	- Method getNumLines() was fixed.
	- A bug related to page break in multi-column mode was fixed.

5.9.108 (2011-08-09)
	- A bug on PHP4 version was fixed.

5.9.107 (2011-08-08)
	- This version includes a minor bugfix.

5.9.106 (2011-08-04)
	- This version includes transparency groups: check the new parameter on startTemplate() method and example 62.

5.9.105 (2011-08-04)
	- Bug item #3386153 "Check Box not ticked when set to true" was fixed.

5.9.104 (2011-08-01)

pagamento.php
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />

<?php 
	require_once("headerbasic.php");
	include("header.php");
	include("lang/Globals_lang_it.php");
	
	//prelevo le variabili del form
	$nome_utente = $_POST["nome"];
	$cognome_utente = $_POST["cognome"];
	$nickname = $_POST["nickname"];
	$is_society = $_POST["is_society"];
	$is_giovane = $_POST["is_giovane"];
	$codfiscale = $_POST["codfiscale"];
	$alternative_url = $_POST["alternative_url"];
	
	//Se è società non può essere giovane
	if($is_society==1)
		$is_giovane==0;
	
	if($is_society==0&&$nickname=="")
		$username = correggi($nome_utente).correggi($cognome_utente);
	else
		$username = correggi($nickname);
	
	$username = strtolower($username);
	$email =  strtolower($_POST["email"]);
	$username_referente = $_POST["username_referente"];
	$password = $_POST["pass"];
	
	$result = Ctrlusernotexists($username);
	
	if($result==false){
		$error_message= "Esiste già un membro con Username: ".$username.", ripeti la registrazione cambiando Username.";
	}else{
		if($username_referente!="tmg"){
			$id_referente = Getidreferente($username_referente);
		}else{
			$id_referente = 0;
			$username_referente = "TOPMANAGERGROUP";
		}
	}
	
	/*$result = Ctrlemailnotexists($email);
	if($result==false)
		$error_message = "<br/>Esiste già un membro con email: ".$email.", ripeti la registrazione cambiando Email.";
	
	
	if($is_giovane==1){
		$result = Ctrlcodfisnotexists($codfiscale);
		if($result==false){
			$error_message = "<br/>Esiste già un membro con Cod fiscale: ".$codfiscale.", Puoi creare solo una card associata ad un unico codice fiscale.";
		}
	}*/
	
	if($error_message==""){
		if(($username!= "")&&($email != "") && ($password != "") ){
			$mysqli = new mysqli(DB_HOST, DB_USER, DB_PASSWORD, DB_DATABASE);
		
			/* check connection */
			if (mysqli_connect_errno()) {
				$error_message= ERR_DB_CONNECTION;
			}
            //$md5password = md5($password);
			$idTransazione =  Generate_IDT();
			/* create a prepared statement */
			$stmt = $mysqli->prepare("INSERT INTO ".USER_TEMP_TABLE." (Username,Nome,Cognome,Password,Email,ID_Referente,idTransazione,society,is_giovane,codfiscale,alternative_url) VALUES (?,?,?,?,?,?,?,?,?,?,?)");
			/* bind parameters for markers */
			$stmt->bind_param("sssssisiiss",$username,$nome_utente,$cognome_utente,$password,$email,$id_referente,$idTransazione,$is_society,$is_giovane,$codfiscale,$alternative_url);
			$stmt->execute();
			$stmt->close();
			/* close connection */
			$mysqli->close();
		}
		else{
			$error_message=ERR_REGISTRATION_DATA;
		}
	}
	
	function Generate_IDT(){
		$N_Caratteri = 16;
		$Stringa = "";
		for($I=0;$I<$N_Caratteri;$I++){
			do{
				$N = ceil(rand(48,122));
			}while(!((($N >= 48) && ($N <= 57)) || (($N >= 65) && ($N <= 90)) || (($N >= 97) && ($N <= 122))));
			$Stringa = $Stringa.chr($N);
		}
		return $Stringa;
	}
	
	function correggi($stringa) {
		$correzioni = array("À" => "A", "Á" => "A", "Â" => "A", "Ä" => "A", "Å" => "A", "Æ" => "AE", "Ç" => "C", "È" => "E", "É" => "E", "Ê" => "E", "Ë" => "E", "Ì" => "I", "Í" => "I", "Î" => "I", "Ï" => "I", "Ñ" => "N", "Ò" => "O", "Ó" => "O", "Ô" => "O", "Ö" => "O", "Ù" => "U", "Ú" => "U", "Û" => "U", "Ü" => "U", "ß" => "ss", "à" => "a", "á" => "a", "â" => "a", "ä" => "a", "æ" => "ae", "ç" => "c", "è" => "e", "é" => "e", "ê" => "e", "ë" => "e", "ì" => "i", "í" => "i", "î" => "i", "ï" => "i", "ñ" => "n", "ò" => "o", "ó" => "o", "ô" => "o", "ö" => "o", "ù" => "u", "ú" => "u", "û" => "u", "ü" => "u", "Ő" => "O", "ő" => "o", "Œ" => "OE", "œ" => "oe");
		foreach($correzioni as $chiave => $valore) {
			$stringa = str_replace($chiave, $valore, $stringa);
		}
		$stringa = eregi_replace("[[:space:]]" , "", $stringa);
		
		//$stringa = eregi_replace("[^a-z0-9._-]" , "", $stringa);
		$stringa = eregi_replace("[^[:alnum:]_-]" , "", $stringa);
		return $stringa;
	} 
	
	function Ctrlusernotexists($username){
		//controllo che l'utente non sia già presente nel DB
		$mysqli = new mysqli(DB_HOST, DB_USER, DB_PASSWORD, DB_DATABASE);
	
		/* check connection */
		if (mysqli_connect_errno()) {
			$error_message= ERR_DB_CONNECTION;
		}
		
		$query = "SELECT Username FROM ".USER_TABLE." WHERE Username = '$username'";
		$result = $mysqli->query($query);
		/* associative array */
		$row = $result->fetch_array(MYSQLI_ASSOC);
		
		/* close connection */
		$result->close();
		$mysqli->close();
		if($row==NULL){
			return true;	
		}else{
			return false;	
		}
		
	}
	
	function Getidreferente($username_referente){
		//controllo che l'utente non sia già presente nel DB
		$mysqli = new mysqli(DB_HOST, DB_USER, DB_PASSWORD, DB_DATABASE);
	
		/* check connection */
		if (mysqli_connect_errno()) {
			$error_message= ERR_DB_CONNECTION;
		}
		
		$query = "SELECT ID FROM ".USER_TABLE." WHERE Username = '$username_referente'";
		$result = $mysqli->query($query);
	
		/* associative array */
		$row = $result->fetch_array(MYSQLI_ASSOC);
		return $row['ID'];
		/* close connection */
		$mysqli->close();
	}

?>


<title>Pagamento - Top Manager Group</title>
<link rel="stylesheet" type="text/css" href="step2/css/step2.css"/>
<link rel="stylesheet" type="text/css" href="css/common.css"/>
<link rel="stylesheet" type="text/css" href="common/css/text.css"/>
<script src="http://cdn.jquerytools.org/1.2.6/full/jquery.tools.min.js"></script>

</head>

<body>

	<div align="center"><a href="index.php"><img src="image/banner/banner_scaled.png" border="0"/></a></div>
	<div align="center">
          <div class="cols">
                      <div style="" class="col">
                      	<?php if($error_message=="" && $is_giovane==0){?>
                                
                            <div align="center" style="text-align:center; margin-top:20px;">
                            	<p class="text14px">Il costo dell'iscrizione annuale a top manager group e' di € 96 annuali incluso i.v.a.</p>
                                  <p class="titolo2">Paga tramite Carta di credito:</p>
                                
                                <form action="https://www.paypal.com/cgi-bin/webscr" method="post">
                                    <input type="hidden" name="cmd" value="_s-xclick">
                                    <input type="hidden" name="hosted_button_id" value="XH65Z8MJHNJ5S">
                                    <input type="image" src="image/icone/logo_carte_credito.png" border="0" name="submit" alt="PayPal - Il metodo rapido, affidabile e innovativo per pagare e farsi pagare.">
                                    <input type="hidden" name="custom" value="<?php echo $idTransazione; ?>">
								</form>                               
                                
                                
                              <!--Sandbox account tmg
                                <form action="https://www.sandbox.paypal.com/cgi-bin/webscr" method="post" target="_top">
                                <input type="hidden" name="cmd" value="_s-xclick">
                                <input type="hidden" name="hosted_button_id" value="C9KUDKV8CFBRQ">
                                <input type="image" src="https://www.sandbox.paypal.com/en_US/i/btn/btn_subscribeCC_LG.gif" border="0" name="submit" alt="PayPal - The safer, easier way to pay online!">
                                <img alt="" border="0" src="https://www.sandbox.paypal.com/en_US/i/scr/pixel.gif" width="1" height="1">
                                <input type="hidden" name="custom" value="<?php echo $idTransazione; ?>">
                                </form>-->
                                 <br/>
                                <p class="titolo2">Paga tramite Paypal:</p>
                                <form action="https://www.paypal.com/cgi-bin/webscr" method="post">
                                    <input type="hidden" name="cmd" value="_s-xclick">
                                    <input type="hidden" name="hosted_button_id" value="XH65Z8MJHNJ5S">
                                    <input type="image" src="https://www.paypalobjects.com/webstatic/mktg/logo-center/logo_paypal_sicuro.png" border="0" name="submit" alt="PayPal - Il metodo rapido, affidabile e innovativo per pagare e farsi pagare.">                                 <br/>
                                    <a href="https://www.paypal.com/it/webapps/mpp/why" target="_blank"><font size="2" face="Arial" color="#0079CD"><strong>Come funziona PayPal</strong></font></a>
                                    <input type="hidden" name="custom" value="<?php echo $idTransazione; ?>">
								</form>
                                 <br/>
                                <p class="titolo2">Paga tramite bonifico bancario:</p>
                                <form action="creabonifico.php" method="post" name="formbonifico">
                                    <input type="hidden" name="idTransazione" value="<?php echo $idTransazione; ?>">
                                    <input type="image" src="image/pagamento/btn/btn_bonifico.png" border="0" name="submit" alt="Paga tramite bonifico."> 
                                    <!--<input type="submit" value="Procedi"/>-->
                                
                            	</form>
                            </div>
                       
                       <?php }
					   else if($error_message=="" && $is_giovane==1){?>
						     <div align="center" style="text-align:center; margin-top:100px;">
                            	<p class="text14px">Complimenti, Hai un'età compresa tra 18 e 24 anni! Puoi usufruire dell'offerta Top Manager Group: La card ti verrà fornita gratuitamente, conferma l'iscrizione: </p>                                <form action="creagiovane.php" method="post">
                            	<input type="hidden" name="idTransazione" value="<?php echo $idTransazione; ?>">
                                <input type="submit" value="Procedi"/>
                            	</form>                  
                                <br/>
                            </div>
                            
					  <? }else{?>
							<div align="center" style="text-align:center; margin-top:150px;">
                            	<p class="titolo2"><? echo $error_message; ?></p>    
                            </div>
					   <?php 
					   }?>
                       </div>
                      <div style="left: 280px;" class="col2">
                       
                       </div>
                      
                      
                      <div style="left: 760px;" class="col3">
                    </div>
               </div>
               
                  
               <div id="card_down">
                  <div style="margin-left:19px;">
                	 <button class="prev" name="submit" type="button" onclick="location.href='index.php?tab=iscrizione'"></button> 
                  </div>
               </div>
           
          </div>

</body>
</html>